src/Game/Game.cpp

```diff
@@ -68,8 +74,47 @@ void Game::initialize()
     auto const transparent = Entity::create("transparent_grass");
     std::shared_ptr<Sprite> const grass = transparent->add_component<Sprite>(Sprite::create(ui_material, "./res/textures/grass.png"));
     transparent->transform->set_local_scale(glm::vec3(0.2f, 0.2f, 0.2f));
-
+    
     auto const model2 = Entity::create("testmodel2");
     model2->add_component(Model::create("./res/models/pyramid3/scene.gltf", standard_material));
     model2->transform->set_parent(model->transform);
+    
+
+    auto const light2 = Entity::create("Point light");
+    Sphere sph = Sphere(0.1f, 10, 10, "./res/textures/container.png", light_source_material);
+    std::shared_ptr<Sphere> sph_ptr = std::make_shared<Sphere>(sph);
+    light2->add_component(sph_ptr);
+    light2->transform->set_local_position(glm::vec3(2.0f, 2.0f, 2.0f));
+    light2->add_component<PointLight>(PointLight::create());
+    light2->get_component<PointLight>()->ambient = glm::vec3(0.05f);
+    light2->get_component<PointLight>()->diffuse = glm::vec3(0.2f);
+    light2->get_component<PointLight>()->specular = glm::vec3(0.2f);
+    light2->get_component<PointLight>()->constant = 1.0f;
+    light2->get_component<PointLight>()->linear = 0.09f;
+    light2->get_component<PointLight>()->quadratic = 0.032f;
+
+   
+    auto const light4 = Entity::create("Spot light");
+    light4->add_component(sph_ptr);
+    light4->transform->set_local_position(glm::vec3(0.0f, 0.0f, -10.0f));
+    light4->transform->set_euler_angles(glm::vec3(180.0f, 0.0f, 0.0f));
+
+    light4->add_component<SpotLight>(SpotLight::create());
+    light4->get_component<SpotLight>()->ambient = glm::vec3(0.0f);
+    light4->get_component<SpotLight>()->diffuse = glm::vec3(1.0f);
+    light4->get_component<SpotLight>()->specular = glm::vec3(1.0f);
+    light4->get_component<SpotLight>()->constant = 1.0f;
+    light4->get_component<SpotLight>()->linear = 0.09f;
+    light4->get_component<SpotLight>()->quadratic = 0.032f;
+    light4->get_component<SpotLight>()->cut_off = glm::cos(glm::radians(12.5f));
+    light4->get_component<SpotLight>()->outer_cut_off = glm::cos(glm::radians(15.0f));
```

Comment:
Holy... this is a lot of `get_component` (which is basically as slow or most likely slower than in Unity), we can just save the `shared_ptr` that `add_component` is returning for us and reuse it!